### PR TITLE
Github Actions for OpenWPM

### PR DESCRIPTION
**Pull request by vringar**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 
 OpenWPM
-[![Build Status](https://travis-ci.org/mozilla/OpenWPM.svg?branch=master)](https://travis-ci.org/mozilla/OpenWPM)
+[![Build Status](https://github.com/vringar/openwpm/workflows/Pre-commit/badge.svg)](https://travis-ci.org/mozilla/OpenWPM)
```
Comment (vringar) [inline README.md:3]: OpenWPM instead of vringar

File: environment.yaml [hunk 2]
```diff
@@ -41,3 +41,6 @@ dependencies:
   - plyvel==1.3.0
   - subprocess32==3.5.4
 name: openwpm
+
+
+
```
Comment (englehardt) [inline environment.yaml:46]:
```suggestion
```

File: test/test_s3_aggregator.py [hunk 3]
```diff
@@ -46,8 +46,8 @@ def get_config(self, num_browsers=1, data_dir=""):
         return manager_params, browser_params
 
     @pytest.mark.skipif(
-        "TRAVIS" in os.environ and os.environ["TRAVIS"] == "true",
-        reason="Localstack fails to start on Travis",
+        "CI" in os.environ and os.environ["CI"] == "true",
+        reason="Localstack fails to start on CI",
```
Comment (boolean5) [inline test/test_s3_aggregator.py:50]: There are two more references to `"TRAVIS"` that we should update in [`test_extension.py`](https://github.com/vringar/OpenWPM/blob/actions/test/test_extension.py#L348L351) and [`test_mp_logger.py`](https://github.com/vringar/OpenWPM/blob/actions/test/test_mp_logger.py#L141L144)

**Review by boolean5 — CHANGES_REQUESTED** [review]

**Comment by boolean5** [comment]
Shouldn't we add a job that will build the docker image and deploy it to dockerhub? Of course this can be done in a subsequent PR.

**Comment by vringar** [comment]
@motin set up the build on TravisCI and I haven't reached out to him for the credentials

**Comment by boolean5** [comment]
Sure, here it is: https://github.com/mozilla/OpenWPM/issues/848

File: test/test_s3_aggregator.py (hunk 3 above)
Comment (vringar) [reply test/test_s3_aggregator.py:50]: I changed those references

Comment (vringar) [reply test/test_s3_aggregator.py:50]: Thanks for catching this.

File: .github/workflows/run-tests.yaml [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# This workflow will run all tests as well as pre-commit 
+
+name: Tests and linting
+on:
+  [push, pull_request]
```
Comment (boolean5) [inline .github/workflows/run-tests.yaml:5]: I just noticed that when using Travis there was a [cron job](https://travis-ci.org/github/mozilla/OpenWPM/builds) running every 2 days. If we want to keep doing this I think something like the following would work:

```suggestion
on:
  push:
  pull_request:
  schedule:
    - cron: '0 0 */2 * *'
```

File: .travis.yml [hunk 5]
```diff
@@ -1,50 +0,0 @@
...
-  - if [[ "$TESTS" != "webextension" ]]; then
-      pre-commit run --all;
-    fi
-script:
-  - ./scripts/travis.sh
-after_success:
-  - if [[ "$TESTS" == "webextension" ]]; then
-      npm run cov:check;
-      npm run cov:send;
-    fi
```
Comment (boolean5) [inline .travis.yml:35]: Should we port this to the Github Actions workflow?

File: .github/workflows/run-tests.yaml [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+    - run: echo "$MINICONDA_PATH/bin" >> $GITHUB_PATH
+
+    - name: Install.sh (cache miss)
+      if: ${{ steps.cache.outputs.cache-hit != 'true' }}
+      run: |
+        $GITHUB_WORKSPACE/install.sh
+    - name: Install.sh (cache hit)
+      if: ${{ steps.cache.outputs.cache-hit == 'true' }}
+      run: |
+        conda info --envs
```
Comment (boolean5) [inline .github/workflows/run-tests.yaml:39]: The `|` is not necessary for single-line commands, right?

**Review by boolean5 — APPROVED** [review]
LGTM. I just added some suggestions for minor changes.

File: .travis.yml (hunk 5 above)
Comment (vringar) [reply .travis.yml:35]: Done!

File: .github/workflows/run-tests.yaml (hunk 6 above)
Comment (vringar) [reply .github/workflows/run-tests.yaml]: Yes, I have removed it :)

**Pull request merged**